### PR TITLE
Result of split_string() can be empty

### PR DESCRIPTION
**Pull request by danpoe**
Bugfix

### CODE REVIEW

File: src/util/string_utils.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ std::string strip_string(const std::string &s)
     return "";
 
   std::string::const_reverse_iterator r_it;
-  for(r_it=s.rbegin(); std::isspace(*r_it); r_it++);
+  for(r_it=s.rbegin(); std::isspace(*r_it); r_it++) {}
```
Comment (tautschnig) [inline src/util/string_utils.cpp]: I'm starting to wonder whether you might want to use `first_not_of("\n\t ")` and `last_not_of("\n\t ")` ?

File: src/util/string_utils.cpp [hunk 2]
```diff
@@ -79,9 +79,9 @@ void split_string(
   unsigned start=0;
   unsigned i;
 
-  for (i=0; i<n; i++)
+  for(i=0; i<n; i++)
   {
-    if (s[i]==delim)
+    if(s[i]==delim)
```
Comment (tautschnig) [inline src/util/string_utils.cpp]: While you are at it: get rid of `unsigned` and use `std::string::size_type` in all those places.

File: src/util/string_utils.cpp (hunk 1 above)
Comment (danpoe) [inline src/util/string_utils.cpp]: Great. I'm using ``find_if_not()`` now.

File: src/util/string_utils.cpp [hunk 3]
```diff
@@ -25,22 +26,18 @@ Author: Daniel Poetzl
...
-  // find first non-space char
-  unsigned i;
-  for(i=0; i<n; i++)
-  {
-    if(!std::isspace(s[i]))
-      break;
-  }
-  if(i==n)
+  std::string::const_iterator left
+    =std::find_if_not(s.begin(), s.end(), pred);
```
Comment (tautschnig) [inline src/util/string_utils.cpp]: Nit picking (applies here and for the other case of std::find_if_not): place the "=" at the end of the previous line, or maybe even use `auto left=...` and put it all on a single line.

**Pull request merged**